services/motion/builtin/move_request.go

```diff
@@ -837,3 +842,28 @@ func toGeoPosePlanSteps(posesByComponent []motionplan.PlanStep, geoPoses []spati
 	}
 	return steps, nil
 }
+
+// This function will check the orientation of the startPose of a 2d move request, and ensure that it is normal to the XY plane. If it is
+// not, it will be altered such that it is (accounting for e.g. an ourdoor base with one wheel on a rock). If the orientation is such that
+// the robot is pointed directly up or down (or is upside-down), an error is returned.
+func correctStartPose(startPose spatialmath.Pose) (spatialmath.Pose, error) {
```

Comment:
Shouldn't this be a filter on the localizer instead of just being done once for a moveRequest?

Reply:
Good point, I think we probably want that there. Otherwise, CheckPlan will be subject to the same issues.